src/app/item-actions/ActionButtons.tsx

```diff
@@ -25,13 +26,18 @@ import styles from './ActionButtons.m.scss';
 interface ActionButtonProps {
   item: DimItem;
   label?: boolean;
+  fromCompare?: boolean;
 }
 
-export function CompareActionButton({ item, label }: ActionButtonProps) {
+export function CompareActionButton({ item, label, fromCompare = false }: ActionButtonProps) {
   const dispatch = useDispatch();
+  const session = useSelector(compareSessionSelector);
 
   const openCompare = () => {
     hideItemPopup();
+    if (fromCompare && session) {
+      session.initialItemId = item.id;
```

Comment:
Same as the other comment - you can't mutate the session like this, you have to dispatch an action that creates a new session. Probably a different version of `addCompareItem`.